packages/datastore/src/storage/adapter/IndexedDBAdapter.ts

```diff
@@ -79,6 +78,47 @@ class IndexedDBAdapter implements Adapter {
 		return storeName;
 	}
 
+	private getNamespaceAndModelFromStorename(storeName: string) {
+		const [namespaceName, ...modelNameArr] = storeName.split('_');
+		return {
+			namespaceName,
+			modelName: modelNameArr.join('_'),
+		};
+	}
+
+	private getIndexKeyPath(namespaceName: string, modelName: string): string[] {
+		const keyPath =
+			this.schema.namespaces[namespaceName]?.keys[modelName].primaryKey;
+
+		if (keyPath) {
+			return keyPath;
+		}
+
+		return ['id'];
+	}
+
+	private getIndexKeyValues<T extends PersistentModel>(model: T): string[] {
+		const modelConstructor = Object.getPrototypeOf(model)
+			.constructor as PersistentModelConstructor<T>;
+		const namespaceName = this.namespaceResolver(modelConstructor);
+		const keys = this.getIndexKeyPath(namespaceName, modelConstructor.name);
+
+		const keyValues = keys.map(field => model[field]);
+		return keyValues;
+	}
+
+	private keysEqual(keysA, keysB): boolean {
+		if (keysA.length !== keysB.length) {
+			return false;
+		}
+
+		if (keysA.length === 1) {
+			return keysA[0] === keysB[0];
+		}
+
+		return keysA.every((key, idx) => key === keysB[idx]);
```

Comment:
If both keys lenght are the same, this check is not needed since `.every` will work just fine with a length of 1

```
		if (keysA.length === 1) {
			return keysA[0] === keysB[0];
		}
```

```suggestion
		return keysA.length === keysB.length && keysA.every((key, idx) => key === keysB[idx]);
```

Reply:
I agree. The idea was to do an early return, but it doesn't add much actual value.